### DISCUSSION TITLE
Smoothed-Finite Element

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#20725] [Q&A Modules: Solid mechanics] [answered]
Hi dear MOOSE experts and users, 

I am simulating a large scale geomechanical model including fault. In order to avoid sharp change of the material properties near the fault, I decided to apply phase-field approach for properly modeling. But, as you probably guess, by doing so  much time is needed for convergence. Then, I just found Smoothed Finite Element, which is more or less different in creating shape function in respect to standard FEM implemented in MOOSE, but it has higher convergence rate.
Now, I wondering to know, is there any way to control the shape functions. Since, based on the formulation of the Smoothed FEM, after obtaining the displacement using standard FEM, smoothing process is being performed on elements resulting a smooth matrix. So, I want to know whether it is possible to perform smoothing process in MOOSE.

Thank you in advance

**Comment by GiudGiud** [comment, marked as answer]
Hello

The shape functions are provided by our underlying finite element library, libmesh. So I'd encourage you to ask this on their github.
https://github.com/libMesh/libmesh/discussions
Are you looking to develop this feature?

Guillaume
